code/modules/mob/inventory.dm

```diff
@@ -247,29 +204,13 @@
 	As far as I can tell the proc exists so that mobs with different inventory slots can override
 	the search through all the slots, without having to duplicate the rest of the item dropping.
 */
-// This proc is going to be killed when inventory slots are merged, eta 300 years
-/mob/proc/set_item_unequipped_legacy(obj/W)
-	SHOULD_CALL_PARENT(TRUE)
-	if(W == _back)
-		_back = null
-		update_inv_back(0)
-		return TRUE
-	if(W == _wear_mask)
-		_wear_mask = null
-		update_inv_wear_mask(0)
-		return TRUE
-	return FALSE
-
 /mob/proc/unequip(obj/W)
 	SHOULD_CALL_PARENT(TRUE)
 	if(!istype(W))
```

Comment:
This is where the QDELETED check was that broke item dropping. It looks to me like this should work fine now.